### DISCUSSION TITLE
Configure dns filter plugin in Logstash

### DISCUSSION THREAD

**Discussion by chadlangston** [Security-Onion-Solutions/securityonion#3994] [Unsupported Versions]
Is there a way to modify a Logstash conf file somewhere in Saltstack such that I can configure the pre-installed Logstash dns filter plugin? All the pieces seem to be right in front of me, but I'm having difficulty connecting these last few dots. I found a few different conf files inside the Logstash container. Looking more closely at the salt directories, I tried puttting a dns filter conf in this folder to add it to logstash on the next restart /opt/so/saltstack/local/salt/logstash/pipelines/config/custom/ but that's didn't work. Finally, I ended up adding my filter config to the 9000_output_zeek.conf.jinja and confirmed that it ends up inside the container, but it still doesn't seem to be working. My next best guess is that the field names I am using are not the correct field names and will depend on what the input source is.

filter {
  dns {
    reverse => [ "source.ip", "destination.ip" ]
    action => "replace"
    nameserver => {
      address => ["172.16.0.1"]
    }
  }
}
output {
  if [module] =~ "zeek" and "import" not in [tags] {
    elasticsearch {
      pipeline => "%{module}.%{dataset}"
      hosts => "{{ ES }}"
      index => "so-zeek"
      template_name => "so-zeek"
      template => "/templates/so-zeek-template.json"
      template_overwrite => true
      ssl => true
      ssl_certificate_verification => false
    }
  }

Thank you!

**Comment by weslambert** [comment]
At least one issue is that those fields are not yet present before the events are sent to the output.  The source.ip and destination.ip fields are set in the Ingest Node pipeline(s) that the event gets pushed to when its routed to the ES output.  You 'll need to pre-parse the data to get at least those fields in order for it to work correctly.

**Reply by chadlangston** [reply]
Thanks for pointing me in the right direction Wes! I'll take a look and see what I can figure out :)
